ArbitraryFileIdManager writes to SQLite database when garbage collected

## Description

The `ArbitraryFileIdManager` does some cleanup in its `__del__` method [here](https://github.com/jupyter-server/jupyter_server_fileid/blob/ffd314832343b2d0ebca62925d523125b7cf3ad7/jupyter_server_fileid/manager.py#L387-L392), which involves writing to a SQLite database.
This creates issues in tests (for example [here](https://github.com/jupyterlab/jupyter-collaboration/actions/runs/8603265131/job/23574762141?pr=279)):
```
E   pytest.PytestUnraisableExceptionWarning: Exception ignored in: <function ArbitraryFileIdManager.__del__ at 0x7fa08739f920>
E   
E   Traceback (most recent call last):
E     File "/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/jupyter_server_fileid/manager.py", line 391, in __del__
E       self.con.commit()
E   sqlite3.ProgrammingError: SQLite objects created in a thread can only be used in that same thread. The object was created in thread id 1403275078[56](https://github.com/jupyterlab/jupyter-collaboration/actions/runs/8603265131/job/23574762141?pr=279#step:6:57)256 and this is thread id 140327414023744.
```
In general, I don't think it is a good practice because the object can be garbage-collected from a different thread that the one in which it was created.
A better solution would be to do this in the teardown of the extension.

--- a/jupyter_server_fileid/manager.py
+++ b/jupyter_server_fileid/manager.py
@@ -386,8 +386,14 @@ class ArbitraryFileIdManager(BaseFileIdManager):
         """Cleans up `ArbitraryFileIdManager` by committing any pending
         transactions and closing the connection."""
         if hasattr(self, "con"):
-            self.con.commit()
-            self.con.close()
+            # If garbage collection happens in a different thread than the thread where
+            # the SQLite object was created, committing will fail anyway. We just ignore
+            # the exception if this is the case.
+            try:
+                self.con.commit()
+                self.con.close()
+            except sqlite3.ProgrammingError:
+                pass
 
 
 class LocalFileIdManager(BaseFileIdManager):

